### PR TITLE
Encorporate sklearn benchmark information for refined parameters (Fixes #161)

### LINKED ISSUE #161: Integrate sklearn benchmark information to refine TPOT classifier parameters
We have the results in from the sklearn benchmark, and we have some decent ideas on what parameters should be tuned for each model and in what ranges. Let's incorporate this information into TPOT to help it with the optimization process.

If I don't mention a particular parameter that's already evolvable for a classifier, then that means the parameter can be left as-is.

# AdaBoostClassifier

Fix `n_estimators` at 500; no longer evolvable

Force `learning_rate` in the range `[0.0001, 1.]`

# BernoulliNB

Fix `fit_prior` to `True`; no longer evolvable

# DecisionTreeClassifier

Remove `max_depth` and `max_features` as evolvable parameters; use the sklearn defaults

Add `min_weight_fraction_leaf` as an evolvable  parameter in the range `[0., 0.5]` (float)

# ExtraTreesClassifier

Change `max_features` to be a float in the range `[0., 1.]`

Remove `max_depth` and use the sklearn default

Add `min_weight_fraction_leaf` as an evolvable parameter in the range `[0., 0.5]` (float)

# GradientBoostingClassifier

Force `learning_rate` in the range `[0.0001, 1.]`

Add `max_features` as an evolvable parameter in the range `[0., 1.]` (float)

Add `min_weight_fraction_leaf` as an evolvable parameter in the range `[0., 0.5]` (float)

Add `loss` as an evolvable parameter w/ options `['exponential', 'deviance']`

Remove `max_depth` and use the sklearn default

# KNeighborsClassifier

Add `weights` as an evolvable parameter w/ options `['uniform', 'distance']`

# LinearSVC

Remove `fit_intercept` and `tol` from the evolvable parameters and use the sklearn defaults

Force `C` in the range `[0.0001, 25.]`

Add `penalty` as an evolvable parameter w/ options `['l1', 'l2']`

Add `dual` as an evolvable parameter w/ options `[True, False]`

# LogisticRegression

Force `C` in the range `[0.0001, 50.]`

Add `penalty` as an evolvable parameter w/ options `['l1', 'l2']`

Add `dual` as an evolvable parameter w/ options `[True, False]`

# MultinomialNB

Fix `fit_prior` to `True`; no longer evolvable

# PassiveAggressiveClassifier

Remove `fit_intercept` from the evolvable parameters and fix it to `True`

Force `C` in the range `[0.0001, 1.]`

# RandomForestClassifier

Remove `max_features` from the evolvable parameters and use the sklearn default

Add `min_weight_fraction_leaf` as an evolvable parameter in the range `[0., 0.5]` (float)

# SVC

Remove this classifier entirely from TPOT for now; too many parameters to simultaneously optimize

### PR DESCRIPTION
**Pull request by teaearlgraycold**
## What does this PR do?

Refines the parameters for almost all TPOT pipeline operators as specified in #161.

## Where should the reviewer start?

By inspecting the parameters of each TPOT pipeline operator.

## How should this PR be tested?

With the built in CI suite

## What are the relevant issues?

#161 

## Questions:

- Do the docs need to be updated?

Yes. I've already updated them.

- Does this PR add new (Python) dependencies?

No

### CODE REVIEW

File: tpot/tpot.py [hunk 1]
```diff
@@ -774,9 +749,13 @@ def _knnc(self, input_df, n_neighbors):
...
 
-        return self._train_model_and_predict(input_df, KNeighborsClassifier, n_neighbors=n_neighbors)
+        weights_values = ['uniform', 'distance']
+        weights_selection = weights_values[weights % len(weights_values)]
 
-    def _gradient_boosting(self, input_df, learning_rate, max_depth):
+        return self._train_model_and_predict(input_df, KNeighborsClassifier,
+            n_neighbors=n_neighbors, weights=weights_selection)
+
+    def _gradient_boosting(self, input_df, learning_rate, max_features, min_weight):
```
Comment (rhiever) [inline tpot/tpot.py]: `_gradient_boosting` needs the `loss` parameter as well. See #161.

**Comment by rhiever** [comment]
I'll just merge this and make that fix real quick.
